metis-indexing/src/main/java/eu/europeana/indexing/FullBeanPublisher.java

```diff
@@ -159,4 +256,226 @@ private void publishToSolr(RdfWrapper rdf, FullBeanImpl fullBean) throws Indexin
+    final SolrDocumentList subQueryResults = solrDocuments.stream().filter(predicate)
+        .collect(Collectors.toCollection(SolrDocumentList::new));
+    if (!CollectionUtils.isEmpty(subQueryResults)) {
+      solrDocuments = subQueryResults;
+    }
+    //Return all identifiers found
+    if (!CollectionUtils.isEmpty(solrDocuments)) {
+      return solrDocuments.stream()
+          .map(document -> (String) document.getFieldValue(EdmLabel.EUROPEANA_ID.toString()))
+          .collect(Collectors.toList());
+    }
+    return new ArrayList<>();
+  }
+
+  private Pair<String, Map<String, List<String>>> generateQueryForMatchingFields(
+      RdfWrapper rdfWrapper) {
+    //Collect all required information for heuristics
+    final List<String> identifiers = rdfWrapper.getProviderProxyIdentifiers().stream()
+        .map(Identifier::getString).collect(Collectors.toList());
+    final List<String> titles = rdfWrapper.getProviderProxyTitles().stream().map(Title::getString)
+        .collect(Collectors.toList());
+    final List<String> descriptions = rdfWrapper.getProviderProxyDescriptions().stream()
+        .map(description -> {
+          if (StringUtils.isNotBlank(description.getString())) {
+            return description.getString();
+          } else if (description.getResource() != null && StringUtils
+              .isNotBlank(description.getResource().getResource())) {
+            return description.getResource().getResource();
+          }
+          return null;
+        }).collect(Collectors.toList());
+    final List<String> isShownByList = rdfWrapper.getIsShownByList().stream()
+        .map(IsShownBy::getResource)
+        .collect(Collectors.toList());
+    final List<String> isShownAtList = rdfWrapper.getIsShownAtList().stream()
+        .map(IsShownAt::getResource)
+        .collect(Collectors.toList());
+
+    //Create all lists that need to be combined
+    final Map<String, List<String>> firstMapOfLists = createFirstCombinationGroup(identifiers,
+        titles, descriptions);
+    final Map<String, List<String>> secondMapOfLists = createSecondCombinationGroup(isShownByList,
+        titles, descriptions);
+    final Map<String, List<String>> thirdMapOfLists = createThirdCombinationGroup(isShownByList,
+        identifiers);
+    final Map<String, List<String>> fourthMapOfLists = createFourthGroup(isShownByList,
+        isShownAtList);
+
+    //Combine different list of fields in groups
+    final String firstQueryGroup = generateQueryForFields(firstMapOfLists);
+    final String secondQueryGroup = generateQueryForFields(secondMapOfLists);
+    final String thirdQueryGroup = generateQueryForFields(thirdMapOfLists);
+    final String fourthQueryGroup = generateQueryForFields(fourthMapOfLists);
+
+    //Join all groups
+    final String combinedQuery = Stream
+        .of(firstQueryGroup, secondQueryGroup, thirdQueryGroup, fourthQueryGroup)
+        .filter(StringUtils::isNotBlank).collect(Collectors.joining(" OR "));
```

Comment:
I don't think this works, as the group string is never blank. At minimum, it is equal to '()'.

Reply:
I made a check inside the method generateQueryForFields

Reply:
I don't think this problem is solved yet.